tests/utils.py

```diff
@@ -197,7 +207,10 @@ def check(self, s):
         match = self.line_pattern.match(s)
         if not match:
             return False
-        _, event_name, _, source = match.groups()
+        if self.threads_info:
+            _, _, event_name, _, source = match.groups()
+        else:
+            _, event_name, _, source = match.groups()
```

Comment:
Continuing from the above, `groups` will have `None` for the optional group if it isn't present. Although at this point using `groupdict` and the group names might be better.